packages/composite-checkout-wpcom/src/components/form-field-annotation.tsx

```diff
@@ -0,0 +1,115 @@
+/**
+ * External dependencies
+ */
+import React, { Component, FunctionComponent } from 'react';
+import styled from '@emotion/styled';
+
+type FormFieldAnnotationProps = {
+	formField: Component;
+
+	// Semantic props
+	labelText: string;
+	descriptionText: string;
+	errorMessage: string;
+
+	// Functional props
+	isError: boolean;
+	isDisabled: boolean;
+
+	// Technical props
+	className?: string;
+	labelId: string;
+	descriptionId: string;
+};
+
+export const FormFieldAnnotation: FunctionComponent< FormFieldAnnotationProps > = ( {
+	formField,
+	labelText,
+	labelId,
+	descriptionText,
+	descriptionId,
+	errorMessage,
+	isError,
+	isDisabled,
+	className,
+} ) => (
+	<div className={ className }>
+		{ labelText && (
+			<Label id={ labelId } isDisabled={ isDisabled }>
```

Comment:
Should this have an `htmlFor` attribute as well? We'd need to know the `id` of the form field, but that seems to me like it would be better for accessibility.